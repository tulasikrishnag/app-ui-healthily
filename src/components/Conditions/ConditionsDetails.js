import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import { Typography, Box, Button, InputLabel, Grid } from "@material-ui/core";
import { useHistory } from "react-router-dom";
import Header from "../shared/Header/Header";
const useStyles = makeStyles((theme) => ({
  boxContainer: {
    paddingTop: "100px",
    paddingLeft: "18px",
    paddingRight: "18px",
  },
  typoPadding: {
    paddingBottom: "20px",
    paddingTop: "10px",
  },
  button: {
    padding: "10px 170px 10px 170px",
  },
  imageContainer: {
    paddingLeft: "20px",
    paddingRight: "20px",
  },
}));

export const ConditionsDetails = (props) => {
  const classes = useStyles();

  const conditionsDetails = useSelector((state) => state.conditions.conditions);
  const [filterData, setFilterData] = useState({});

  useEffect(() => {
    const searchData = conditionsDetails.find((el) => {
      return el.label === props.match.params.label;
    });
    setFilterData(searchData);
  }, [conditionsDetails, props.match.params.label]);
  const { snippet, label, image,synonyms } = filterData;
  const history = useHistory();
  return (
   
    <Grid
      container
      spacing={2}
      direction="row"
      alignItems="center"
      justify="center"
    >
      <Header />
      <Box className={classes.boxContainer} maxWidth="600px" display="row">
        <Box className={classes.imageContainer}>
          <img src={image} alt="Failed to load" />

          <InputLabel>Label</InputLabel>
          <Typography className={classes.typoPadding}>{label}</Typography>
          <InputLabel>Snippet</InputLabel>
          <Typography className={classes.typoPadding}>{snippet}</Typography>
          <InputLabel>Synonyms</InputLabel>
          <Typography className={classes.typoPadding}>{synonyms}</Typography>
        </Box>

        <Box className={classes.button}>
          <Button
            variant="contained"
            color="primary"
            onClick={() => history.goBack()}
            data-testid="back-button-test-id"
          >
            BACK
          </Button>
        </Box>
      </Box>
    </Grid>
  
  );
};
