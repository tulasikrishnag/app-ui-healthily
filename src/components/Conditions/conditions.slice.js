import { createSlice } from "@reduxjs/toolkit";

const initialState = {
  conditions: [],
};

export const conditionsSlice = createSlice({
  name: "conditions",
  initialState,
  reducers: {
    pushConditionsToStore: (state, action) => {
      state.conditions.push(...action.payload);
    },
  },
});

export const { pushConditionsToStore } = conditionsSlice.actions;
