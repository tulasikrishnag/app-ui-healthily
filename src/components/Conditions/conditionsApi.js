import axios from "axios";
import * as constants from '../constants'
export const conditionsApi = {
  getConditions: async () => {
    try {
    
      const response = await axios.get(
        `${constants.BACK_END_URL}:${constants.BACK_END_PORT}/conditions`
      );
      return response.data;
    } catch (error) {
      console.log(error);
      throw error;
    }
  },
};
