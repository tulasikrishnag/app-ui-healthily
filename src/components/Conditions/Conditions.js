import React, { useState, useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";
import {
  Box,
  Grid,
  Typography,
  Card,
  CircularProgress,
  Link,
} from "@material-ui/core";
import Header from "../shared/Header/Header";
import { useDispatch } from "react-redux";
import { pushConditionsToStore } from "./conditions.slice";
import { conditionsApi } from "./conditionsApi";
import { useSelector } from "react-redux";
import { withRouter, useHistory } from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    padding: theme.spacing(2),
    [theme.breakpoints.down("sm")]: {
      backgroundColor: "#EFF5FB",
    },
    [theme.breakpoints.up("md")]: {
      backgroundColor: "#F2F2F2",
    },
    [theme.breakpoints.up("lg")]: {
      backgroundColor: "#E0F8F7",
    },
  },
  imageSize: {
    maxWidth: "100%",
    height: "auto",
  },

  card: {
    paddingTop: "10px",
    backgroundColor: "#FAFAFA",
  },
  spinner: {
    paddingTop: "300px",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: "inherit",
    backgroundColor: "white",
  },
}));

const ConditionsUI = (props) => {
  const classes = useStyles();
  const history = useHistory();
  const {label,snippet,image} = props.conditionDetails
  return (
    <Box className={classes.boxContainer}>
      <Card className={classes.card}>
        <img
          src={image}
          className={classes.imageSize}
          alt="Failed to load"
        />

        <Typography>Title: {label}</Typography>
        <Typography noWrap>{snippet}</Typography>
        <Link onClick={() => history.push(`/conditionDetails/${label}`)}>
          Show more
        </Link>
      </Card>
    </Box>
  );
};

const Conditions = () => {
  const classes = useStyles();
  const [conditionsData, setConditionsData] = useState([]);
  const [showLoadingSpinner, setShowLoadingSpinner] = useState(true);

  const dispatch = useDispatch();

  const conditionsDetails = useSelector((state) => state.conditions.conditions);

  useEffect(() => {
    const fetchConditions = async () => {
      const response = await conditionsApi.getConditions();
      dispatch(pushConditionsToStore(response.conditions));
      setConditionsData(response.conditions);
    };
    if (conditionsDetails.length === 0) {
      fetchConditions();
    } else {
      setConditionsData(conditionsDetails);
    }
    const loaderTimeout = setTimeout(() => setShowLoadingSpinner(false), 1000);

    return () => {
      window.clearTimeout(loaderTimeout);
    };
  }, [dispatch, conditionsDetails]);

  return (
    <>
      <Header />
      {showLoadingSpinner ? (
        <div className={classes.spinner}>
          <CircularProgress />
        </div>
      ) : (
        <div className={classes.root}>
          <Grid
            container
            spacing={2}
            direction="row"
            alignItems="flex-start"
            justify="flex-start"
          >
            {conditionsData.map(
              (el) =>
                el.image !== undefined &&
                el.snippet !== undefined && (
                  <Grid
                    item
                    xs={12}
                    sm={6}
                    md={4}
                    key={conditionsData.indexOf(el)}
                  >
                    <ConditionsUI
                      conditionDetails={{image:el.image,snippet:el.snippet,label:el.label}}
                     
                    />
                  </Grid>
                )
            )}
          </Grid>
        </div>
      )}
    </>
  );
};

export default withRouter(Conditions);
