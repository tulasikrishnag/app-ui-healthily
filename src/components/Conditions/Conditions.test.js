import { render, fireEvent } from "@testing-library/react";
import Conditions from "./Conditions";
import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";

describe("Conditions", () => {
  const initialState = {
    conditions: {
      conditions: [
        {
          label: "Acanthosis nigricans",
          synonyms: ["Acquired acanthosis nigricans", "Keratosis nigricans"],
        },
        {
          snippet:
            "Addisonâ€™s disease (also known as primary adrenal insufficiency or hypoadrenalism) is a rare disorder of the adrenal glands.",
          label: "Addison's disease",
          synonyms: [
            "Addison disease",
            "Primary adrenocortical insufficiency",
            "Primary hypoadrenalism",
            "Adison's disease",
            "hypocorticism",
            "adrenal cortex dysplasia",
            "Bronzed disease",
            "adrenal cortex dysfunction",
            "adrenal cortex insufficiency",
            "Melasma addisonii",
            "Adrenal storm",
            "Primary adrenal deficiency",
            "Suppression adrenal",
            "Adrenal Suppresion",
            "adrenal insufficiency",
          ],
          keywords: [
            "addisons disease",
            "hypoadrenalism",
            "primary adrenal insufficiency",
          ],
        },
        {
          snippet:
            "Alzheimer's disease is the most common cause of dementia. Dementia is a group of symptoms associated with a decline in the way your brain functions, affecting your memory and the way you behave.",
          label: "Alzheimer's disease",
          synonyms: ["AD", "Alzheimer disease", "Alzheimers disease"],
          keywords: ["Alzheimer's disease", "dementia"],
          image:
            "http://assets.your.md/conditions/alzheimers-disease/card/alzheimers-disease-male-card.jpg",
        },
        {
          label: "Sickle cell anaemia",
          synonyms: ["Sickle", "sickle cell"],
        },
      ],
    },
  };
  const mockStore = configureStore();
  let store;
  it("render component as expected", () => {
    store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Conditions />
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
