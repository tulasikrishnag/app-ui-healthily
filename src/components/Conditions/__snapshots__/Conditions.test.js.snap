// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conditions render component as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-bg-5 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h4
          class="MuiTypography-root makeStyles-tg-6 MuiTypography-h4"
        >
          Healthily
        </h4>
      </div>
    </header>
  </div>
  <div
    class="makeStyles-spinner-4"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
